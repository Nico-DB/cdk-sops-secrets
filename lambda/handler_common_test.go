package main

import (
	"context"
	"encoding/json"
	"testing"
	"time"

	"github.com/aws/aws-lambda-go/cfn"
	"github.com/aws/aws-lambda-go/lambdacontext"
	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_FullWorkflow_Delete(t *testing.T) {
	mocks := &AWS{
		secretsmanager: &SecretsManagerMockClient{
			t: t,
		},
		s3downlaoder: &S3ManagerMockClient{
			t: t,
		},
	}
	d := time.Now().Add(50 * time.Millisecond)
	ctx, _ := context.WithDeadline(context.Background(), d)
	ctx = lambdacontext.NewContext(ctx, &lambdacontext.LambdaContext{
		AwsRequestID:       "AwsRequestID",
		InvokedFunctionArn: "arn:aws:lambda:us-east-2:123456789012:function:blank-go",
	})
	inputJson := ReadJSONFromFile(t, "events/event_delete.json")
	var event cfn.Event
	err := json.Unmarshal(inputJson, &event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}
