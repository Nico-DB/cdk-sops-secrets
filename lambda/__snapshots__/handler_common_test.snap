
[Test_FullWorkflow_Delete - 1]
>>>syncSopsToSecretsmanager

map[string]interface {}{}
nil
---
